Mathlib/Data/SetLike/SMul.lean

```diff
@@ -0,0 +1,19 @@
+
+/-
+Copyright (c) 2025 Kevin Buzzard. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard
+-/
+import Mathlib.Algebra.Group.Operations
+import Mathlib.Data.SetLike.Basic
+
+/-!
+# Instance for scalar multiplication of set-like objects
+
+-/
+
+variable (R A : Type*) [SMul R A] (S : Type*) [SetLike S R] (s : S)
+
+@[to_additive]
+instance : SMul s A where
```

Comment:
Can you add the `MulAction`, `DistribMulAction` etc versions too for `SubmonoidClass S R`?

Reply:
If by "etc" you mean "and `MulDistribMulAction`" then done, but if there are more I'm unaware of then I'm not done yet!

Reply:
Regarding `MulAction` for `SubmonoidClass`, see #11304 which (a) shows various places to remove redundant instances but also (b) ran into performance issues (though I tend to think such a change would be a good idea even with performance issues).

Reply:
Many thanks for that very pertinent pointer Joseph! In the interim I found `MulActionWithZero` BTW

Reply:
Oh, we don't have `SubmonoidWithZero` or `SubmonoidWithZeroClass` so this is future work.